### PR TITLE
Improve backgroundmode and running in service

### PR DESCRIPTION
**Pull request by williamrijksen**
- [x] Continue scanning onDestroy of Activity when running in a service. It can be enabled by calling `self.Beacons.setRunInService(true)`. `unBindService` is necessary on `taskremoved` inside a started service.
- [x] Change backgroundmode `onPause` and `onResume`
- [x] Optimize javacode

### CODE REVIEW

File: android/manifest [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # this is your module manifest and used by Titanium
 # during compilation, packaging, distribution, etc.
 #
-version: 1.2.1
+version: 1.2.2
```
Comment (jvandijk) [inline android/manifest]: This is not a patch upgrade. You are definitely adding features here...

**Comment by jvandijk** [comment]
Keep an eye on the single commit responsibility. Because of all the code style changes I have difficulties verifying this commit which is easily solvable with single commit responsibility.

**Comment by jvandijk** [comment]
Please update both documentation files as well

File: android/manifest (hunk 1 above)
Comment (williamrijksen) [inline android/manifest]: @jvandijk Do you expect a version like 1.3.0 in this case?

Comment (jvandijk) [inline android/manifest]: Yes

**Comment by williamrijksen** [comment]
@jvandijk Version number and documentation files updated now.

**Comment by jvandijk** [comment]
Nice! Merging

**Pull request merged**